identity/userpass.go

```diff
@@ -26,12 +26,19 @@ func (u *UserPass) Auth(creds *Credentials) (*AuthDetails, error) {
 	if u.client == nil {
 		u.client = goosehttp.New()
 	}
+	// In Keystone v2 TenantName and TenantID can be interchangeable used.
 	auth := authWrapper{Auth: authRequest{
 		PasswordCredentials: passwordCredentials{
 			Username: creds.User,
 			Password: creds.Secrets,
 		},
-		TenantName: creds.TenantName}}
+	}}
+
+	if creds.TenantName == "" && creds.TenantID != "" {
```

Comment:
Retain the deleted line. Then just do this:
if auth.Auth.TenantName == "" {
auth.Auth.TenantName = creds.TenantID
}